internal/webhooks/cluster.go

```diff
@@ -395,6 +436,183 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 	return allWarnings, allErrs
 }
 
+func validateControlPlaneVersion(ctx context.Context, fldPath *field.Path, ctrlClient client.Reader, oldCluster *clusterv1.Cluster, oldVersion semver.Version) *field.Error {
```

Comment:
```suggestion
func validateControlPlaneVersion(ctx context.Context, fldPath *field.Path, ctrlClient client.Reader, newCluster *clusterv1.Cluster, oldVersion semver.Version) *field.Error {
```
Let's use the newCluster in cases where it doesn't really matter